src/main/java/com/google/gcs/sdrs/dao/impl/PooledStsJobDaoImpl.java

```diff
@@ -63,7 +63,7 @@ public PooledStsJobDaoImpl() {
   }
 
   @Override
-  public PooledStsJob getJob(
+  public List<PooledStsJob> getJobList(
```

Comment:
better to create a new method than to replace the existing one.

Reply:
can you explain the logic on how to get the list?

Reply:
So the old logic is to try to pick the job after current schedule time. If no jobs found, it will pick the last job of the day. I kept the behavior unchanged and try to filter all jobs with the same schedule time instead of just return one job.
The method has two parts: 1. pull jobs from database by project, bucket, and type. 2. Pick a job(s) based on the schedule time. Maybe it's better to separate the `pick job` from this and put it to non-dao class. It's also easier to extend pick job strategy in the future.

Reply:
To be safe, the new getJobList method should only be called by dataset execution logic as only the dataset retention processing would use multiple jobs.

Reply:
public TransferJob findPooledJob is the one to keep compatibility. I guess all the execution relied on it before. It's hard to keep the new logic only used new getJobList unless we want to keep redundant code.